src/vs/editor/common/modes.ts

```diff
@@ -392,7 +392,7 @@ export interface ReferenceProvider {
 	/**
 	 * Provide a set of project-wide references for the given position and document.
 	 */
-	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, token: CancellationToken): Location[] | Thenable<Location[]>;
+	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, progress: (value: Location) => void, token: CancellationToken): Location[] | Thenable<Location[]>;
```

Comment:
@felixfbecker I can think of two options (and I don't know which one is best at this point).

1. Accept a type union of `Location` and a new type `PartialLocation` (or some similar name).
2. Only accept `Location` with the understanding that some fields may be blank and will get updated later.

Do you have a proposal?

For this particular PR, I think `Location` is the right type.